MovieViewer/MoviesViewController.swift

```diff
@@ -287,11 +301,126 @@ extension MoviesViewController: UITableViewDataSource {
         
         let cell = tableView.dequeueReusableCellWithIdentifier("MovieCell", forIndexPath: indexPath) as! MovieCell
         let movie = filteredMovies![indexPath.row]
+        
         cell.setData(movie)
         cell.setTheme()
         
+        // retrieve all favorites by id
+        self.ref!.child("favorite/\(movie.id!)").observeEventType(FIRDataEventType.Value, withBlock: { (snapshot) in
```

Comment:
you're typing `favorite/\(movie.id!)` a lot, maybe move that to a property on the model? like `firebasePath` or something?